test/tc_server_tests.cpp

```diff
@@ -1175,3 +1176,30 @@ TEST(TASK_CONTROLLER_SERVER_TESTS, MessageEncoding)
 	}
 	CANHardwareInterface::stop();
 }
+
+TEST(TASK_CONTROLLER_SERVER_TESTS, DDOPHelper)
+{
+	DeviceDescriptorObjectPool ddop(3);
+	ddop.deserialize_binary_object_pool(testDDOP, sizeof(testDDOP));
+
+	auto implement = DeviceDescriptorObjectPoolHelper::get_implement_geometry(ddop);
+
+	ASSERT_EQ(1, implement.booms.size());
+	ASSERT_EQ(16, implement.booms.at(0).sections.size());
+	EXPECT_TRUE(implement.booms.at(0).subBooms.empty());
+
+	EXPECT_TRUE(implement.booms.at(0).xOffset_mm);
+	EXPECT_TRUE(implement.booms.at(0).yOffset_mm);
+	EXPECT_TRUE(implement.booms.at(0).zOffset_mm);
+
+	for (std::size_t i = 0; i < 16; i++)
+	{
+		EXPECT_TRUE(implement.booms.at(0).sections.at(i).width_mm);
+		EXPECT_TRUE(implement.booms.at(0).sections.at(i).xOffset_mm);
+		EXPECT_TRUE(implement.booms.at(0).sections.at(i).yOffset_mm);
+		EXPECT_FALSE(implement.booms.at(0).sections.at(i).zOffset_mm);
+
+		EXPECT_NE(0, implement.booms.at(0).sections.at(i).width_mm.get());
+		EXPECT_NE(0, implement.booms.at(0).sections.at(i).yOffset_mm.get());
```

Comment:
suggestion(non-blocking): probably better to match the actual expected values here rather than an arbitrary number not equal to 0